pkg/telemetry/client_test.go

```diff
@@ -0,0 +1,454 @@
+	configtypes "github.com/vmware-tanzu/tanzu-plugin-runtime/config/types"
+)
+
+func TestClient(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Telemetry plugin test suite")
+}
+
+type mockMetricsDB struct {
+	createSchemaCalled             bool
+	saveOperationMetricCalled      bool
+	createSchemaReturnError        error
+	saveOperationMetricReturnError error
+}
+
+func (mc *mockMetricsDB) CreateSchema() error {
+	mc.createSchemaCalled = true
+	return mc.createSchemaReturnError
+}
+
+func (mc *mockMetricsDB) SaveOperationMetric(payload *OperationMetricsPayload) error {
+	mc.saveOperationMetricCalled = true
+	return mc.saveOperationMetricReturnError
+}
+
+var _ = Describe("Unit tests for UpdateCmdPreRunMetrics()", func() {
+	const True = "true"
+	var (
+		tc           *telemetryClient
+		metricsDB    *mockMetricsDB
+		rootCmd      *cobra.Command
+		cmd          *cobra.Command
+		configFile   *os.File
+		configFileNG *os.File
+		err          error
+	)
+
+	BeforeEach(func() {
+		metricsDB = &mockMetricsDB{}
+		tc = &telemetryClient{
+			currentOperationMetrics: &OperationMetricsPayload{
+				StartTime: time.Time{},
+			},
+			metricsDB: metricsDB,
+		}
+
+		rootCmd = &cobra.Command{
+			Use: "tanzu",
+			RunE: func(cmd *cobra.Command, args []string) error {
+				return nil
+			},
+		}
+
+		configFile, err = os.CreateTemp("", "config")
+		Expect(err).To(BeNil())
+		os.Setenv("TANZU_CONFIG", configFile.Name())
+
+		configFileNG, err = os.CreateTemp("", "config_ng")
+		Expect(err).To(BeNil())
+		os.Setenv("TANZU_CONFIG_NEXT_GEN", configFileNG.Name())
+
+		err = configlib.SetCLIId("fake-cli-id")
+		Expect(err).ToNot(HaveOccurred(), "failed to set the CLI ID")
+
+	})
+	AfterEach(func() {
+		os.Unsetenv("TANZU_CONFIG")
+		os.Unsetenv("TANZU_CONFIG_NEXT_GEN")
+
+		os.RemoveAll(configFile.Name())
+		os.RemoveAll(configFileNG.Name())
+
+	})
+	Describe("when the command is CLI core command", func() {
+		BeforeEach(func() {
+			cmd = &cobra.Command{
+				Use: "corecommand",
+				RunE: func(cmd *cobra.Command, args []string) error {
+					return nil
+				},
+			}
+			rootCmd.AddCommand(cmd)
+		})
+		Context("when core command has arguments provided but no flags", func() {
+			It("should return success and the metrics should have name arg(first argument) hashed and flags string should be empty", func() {
+				tc.UpdateCoreCommandMap(cmd)
+				err = tc.UpdateCmdPreRunMetrics(cmd, []string{"arg1"})
+				Expect(err).ToNot(HaveOccurred())
+				metricsPayload := tc.currentOperationMetrics
+				Expect(metricsPayload.CommandName).To(Equal("corecommand"))
+				Expect(metricsPayload.NameArg).To(Equal(hashString("arg1")))
+				Expect(metricsPayload.Flags).To(BeEmpty())
+				Expect(metricsPayload.CliID).ToNot(BeEmpty())
+				Expect(metricsPayload.StartTime.IsZero()).To(BeFalse())
+
+			})
+		})
+		Context("when the core command(requiring flag values hashed) has arguments and flags provided", func() {
```

Comment:
nit: `command (requiring hashed values)`...
similarly for consistency in L145
`command (not requiring hashed values)`...

Reply:
updated